### PR TITLE
Add settings service

### PR DESCRIPTION
**Pull request by oneumyvakin**

### CODE REVIEW

File: settings.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	ComputeResource       SettingsComputeResource       `json:"compute_resource"`
+	NonExistentVMSRemover SettingsNonExistentVMSRemover `json:"non_existent_vms_remover"`
+	BillingIntegration    SettingsBillingIntegration    `json:"billing_integration"`
+	DNS                   SettingsDNS                   `json:"dns"`
+	Update                SettingsUpdate                `json:"update"`
+	Features              SettingsFeatures              `json:"features"`
+}
+
+// SettingsResponse represents response from settings end-point.
+type SettingsResponse struct {
```
Comment (dshemin) [inline settings.go:29]: This type shouldn't be public.

File: settings.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	Notifications         SettingsNotifications         `json:"notifications,omitempty"`
+	ComputeResource       SettingsComputeResource       `json:"compute_resource,omitempty"`
+	NonExistentVMSRemover SettingsNonExistentVMSRemover `json:"non_existent_vms_remover,omitempty"`
+	BillingIntegration    SettingsBillingIntegration    `json:"billing_integration,omitempty"`
+	DNS                   SettingsDNS                   `json:"dns,omitempty"`
+	Update                SettingsUpdate                `json:"update,omitempty"`
+	Features              SettingsFeatures              `json:"features,omitempty"`
+}
+
+type settingsResponse struct {
```
Comment (dshemin) [inline settings.go:51]: Please move this type closer to the methods where it used.

**Review by dshemin — CHANGES_REQUESTED** [review]

File: settings.go (hunk 1 above)
Comment (oneumyvakin) [reply settings.go:29]: OK

File: settings.go (hunk 2 above)
Comment (oneumyvakin) [reply settings.go:51]: OK

File: settings.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+	ServerResetPassword           SettingsNotificationsTemplate `json:"server_reset_password,omitempty"`
+	UserResetPassword             SettingsNotificationsTemplate `json:"user_reset_password,omitempty"`
+	UserVerifyEmail               SettingsNotificationsTemplate `json:"user_verify_email,omitempty"`
+	ProjectUserInvite             SettingsNotificationsTemplate `json:"project_user_invite,omitempty"`
+	ProjectUserLeft               SettingsNotificationsTemplate `json:"project_user_left,omitempty"`
+	ServerIncomingTrafficExceeded SettingsNotificationsTemplate `json:"server_incoming_traffic_exceeded,omitempty"`
+	ServerOutgoingTrafficExceeded SettingsNotificationsTemplate `json:"server_outgoing_traffic_exceeded,omitempty"`
+}
+
+type SettingsNotificationsTemplate struct {
```
Comment (dshemin) [inline settings.go:94]: Please add a comment for this type.

File: settings_test.go [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+	fakeSettingsResponse := struct {
+		Data Settings `json:"data"`
+	}{}
+	require.NoError(t, json.Unmarshal(fakeSettingsRawJSON, &fakeSettingsResponse))
+
+	require.Equal(t, fakeSettingsResponse.Data, actual)
+}
+
+var fakeSettingsRawJSON = []byte(`{
```
Comment (dshemin) [inline settings_test.go:63]: Why we should use a raw JSON instead of out own structure?

File: limitGroup.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package solus
+
+type LimitGroup struct {
```
Comment (dshemin) [inline limitGroup.go:3]: Please add a comment for this type.

File: limitGroup.go [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+	ID                    int             `json:"id"`
+	Name                  string          `json:"name"`
+	CreatedAt             string          `json:"created_at"`
+	UpdatedAt             string          `json:"updated_at"`
+	VirtualServers        LimitGroupLimit `json:"vms"`
+	RunningVirtualServers LimitGroupLimit `json:"running_vms"`
+	AdditionalIPs         LimitGroupLimit `json:"additional_ips"`
+}
+
+type LimitGroupLimit struct {
```
Comment (dshemin) [inline limitGroup.go:13]: Please add a comment for this type.

**Review by dshemin — CHANGES_REQUESTED** [review]

File: limitGroup.go (hunk 6 above)
Comment (oneumyvakin) [reply limitGroup.go:13]: OK

File: limitGroup.go (hunk 5 above)
Comment (oneumyvakin) [reply limitGroup.go:3]: OK

File: settings.go (hunk 3 above)
Comment (oneumyvakin) [reply settings.go:94]: OK

File: settings_test.go (hunk 4 above)
Comment (oneumyvakin) [reply settings_test.go:63]: Raw response from real instance helps to find type mismatch errors. Own structure can't be used for testing JSON unmarshaling.

**Pull request merged**